src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostConfiguration.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Hosting
+{
+    /// <summary>
+    /// WebAssemblyHostConfiguration is a class that implements the interface of an IConfiguration,
+    /// IConfigurationRoot, and IConfigurationBuilder. It can be used to simulatneously build
+    /// and read from a configuration object.
+    /// </summary>
+    public class WebAssemblyHostConfiguration : IConfiguration, IConfigurationRoot, IConfigurationBuilder
+    {
+        private readonly List<IConfigurationProvider> _providers = new List<IConfigurationProvider>();
+        private readonly List<IConfigurationSource> _sources = new List<IConfigurationSource>();
+
+        private readonly List<IDisposable> _changeTokenRegistrations = new List<IDisposable>();
+        private ConfigurationReloadToken _changeToken = new ConfigurationReloadToken();
+
+        /// <summary>
+        /// Returns the sources used to obtain configuration values.
+        /// </summary>
+        public IList<IConfigurationSource> Sources => _sources.ToList();
+
+        /// <summary>
+        /// Returns the providers user to obtain configuration values.
+        /// </summary>
+        public IEnumerable<IConfigurationProvider> Providers => _providers.ToList();
+
+        /// <summary>
+        /// Gets a key/value collection that can be used to share data between the <see cref="IConfigurationBuilder"/>
+        /// and the registered <see cref="IConfigurationProvider"/>s.
+        ///
+        /// In this implementation, this largely exists as a way to satisfy the
+        /// requirements of the IConfigurationBuilder and is not populated by
+        /// <see cref="WebAssemblyHostConfiguration"/> with any meaningful info.
```

Comment:
Since this is on a public type, I suggest reducing these lines to regular `//` comments, outside the XML doc comment block. Customers probably don't need to know this since it's about an implementation detail.